benchmark/index.js

```diff
@@ -0,0 +1,189 @@
+const cdkV2Handler = require("../cdk/v2/handler");
+const logger = require("./metaLogger");
+const { TRANSFORMER_METRIC } = require("../v0/util/constant");
+
+const command = new Commander.Command();
+command
+  .allowUnknownOption()
+  .option(
+    "-d, --destinations <string>",
+    "Enter destination names separated by comma",
+    "algolia,pinterest_tag"
+  )
+  .option("-s, --sources <string>", "Enter source names separated by comma", "")
+  .parse();
+
+const cmdOpts = command.opts();
+
+const testDataDir = path.join(__dirname, "./../__tests__/data");
+const getTestData = (intgList, fileNameSuffix) => {
+  const intgTestSet = {};
+  intgList.forEach(intg => {
+    try {
+      intgTestSet[intg] = JSON.parse(
+        fs.readFileSync(
+          path.join(testDataDir, `${intg}${fileNameSuffix}.json`),
+          {
+            encoding: "utf-8"
+          }
+        )
+      );
+    } catch (err) {
+      logger.error(`Unable to load the data for: ${intg}`);
+      logger.error(`Raw error: ${err}`);
+    }
+  });
+  return intgTestSet;
+};
+
+const destinationsList = cmdOpts.destinations
+  .split(",")
+  .map(x => x.trim())
+  .filter(x => x !== "");
+logger.info("Destinations selected: ", destinationsList);
+const destTestSet = getTestData(destinationsList, "_input");
+
+const nativeDestHandlers = {};
+const destCdKWorkflowEngines = {};
+
+
+const sourcesList = cmdOpts.sources
+  .split(",")
+  .map(x => x.trim())
+  .filter(x => x !== "");
+logger.info("Sources selected: ", sourcesList);
+const srcTestSet = getTestData(sourcesList, "_source");
+
+async function initializeHandlers() {
+  for (const idx in destinationsList) {
+    const dest = destinationsList[idx];
+    nativeDestHandlers[dest] = versionedRouter.getDestHandler("v0", dest);
+    destCdKWorkflowEngines[dest] = await cdkV2Handler.getWorkflowEngine(
+      dest,
+      TRANSFORMER_METRIC.ERROR_AT.PROC
+    );
+  }
+}
+
+async function run() {
+  await initializeHandlers();
+
+  for (const dest in destTestSet) {
+    for (const td in destTestSet[dest]) {
+      const curTcData = destTestSet[dest][td];
+      let tcInput;
+      let tcDesc;
+      if (
+        "description" in curTcData &&
+        "input" in curTcData &&
+        "output" in curTcData
+      ) {
+        tcInput = curTcData.input;
+        tcDesc = `Destination - ${dest} - "${curTcData.description}"`;
+      } else {
+        tcInput = curTcData;
+        tcDesc = `Destination - ${dest} - "${td}"`;
+      }
+
+      await runDataset(tcDesc, tcInput, dest);
+    }
+  }
+
+  for (const src in srcTestSet) {
+    for (const td in srcTestSet[src]) {
+      const curTcData = srcTestSet[src][td];
+      let tcInput;
+      let tcDesc;
+      if (
+        "description" in curTcData &&
+        "input" in curTcData &&
+        "output" in curTcData
+      ) {
+        tcInput = curTcData.input;
+        tcDesc = `Source - ${src} - "${curTcData.description}"`;
+      } else {
+        tcInput = curTcData;
```

Comment:
For the test cases where we don't have new template of input,output in same file, shouldn't we take `curTcData .message` ?